src/wtc-controller-viewports.js

```diff
@@ -57,38 +60,47 @@ class Viewport extends ElementController {
     this._onPlay = this._onPlay.bind(this);
     this._onResize = this._onResize.bind(this);
     this._onTidy = this._onTidy.bind(this);
-    
+
     // Check for tidy up every 5 seconds
     this.tidyInterval = setInterval(this._onTidy, 5000);
 
     // bind the resize handler
-    window.addEventListener('resize', this._onResize);
+    window.addEventListener("resize", this._onResize);
     this._onResize();
 
-    if ('IntersectionObserver' in window) {
+    if ("IntersectionObserver" in window) {
       // create the intersection ovserver
       this.observer = new IntersectionObserver(this._onObserve, {
-          rootMargin: '0%',
-          threshold: [.1]
-        });
+        rootMargin: "0%",
+        threshold: [0.1]
+      });
       this.observer.observe(this.element);
-    }
-    else {
-      console.log('%cIntersection Observers not supported', 'color: red');
+    } else {
+      console.log("%cIntersection Observers not supported", "color: red");
       this.runAnimation(100, 100, 100);
     }
 
-
     // debug element
-    if(this.element.querySelector('.vp-debug')) {
-      this._debugElement = this.element.querySelector('.vp-debug');
+    if (this.element.querySelector(".vp-debug")) {
+      this._debugElement = this.element.querySelector(".vp-debug");
     }
 
     this.element.classList.add(`${this.classPrefix}--initialised`);
 
     // This manually sets the initial top offset to provide a single, initial call of the animation event.
-    window.addEventListener('load', () => {
-      this.top = this.element.offsetTop - window.scrollY;
+    window.addEventListener("load", () => {
+      // Loop through parent nodes of the element
+      // in order to get a top offset which is relative to the document,
+      // as opposed to the immediate offset parent:
+      let element = this.element;
+      let elementOffset = 0;
+
+      while (element !== document.body) {
```

Comment:
Given that it's possible that document.body can be turned into a non-offset parent (or the loop, in some situations, can simply skip over it). We should change this to:
```
while (element !== document.body) {
  elementOffset += element.offsetTop;
  element = element.parentNode;
  if(element === null) break;
}
```